src/server/client-data-mapper.js

```diff
@@ -55,6 +55,38 @@ class ClientDataMapper {
     });
   }
 
+  async mapImportBatches(batches, user) {
+    const idSet = new Set(batches.map(batch => batch.createdBy));
+    const users = await this.userService.getUsersByIds(Array.from(idSet));
+    const allowedUserFields = privateData.getAllowedUserFields(user);
+
+    if (users.length !== idSet.size) {
+      throw new Error(`Was searching for ${idSet.size} users, but found ${users.length}`);
+    }
+
+    const userMap = new Map(users.map(u => [u._id, u]));
+    return batches.map(batch => {
+      return {
+        ...batch,
+        createdBy: this._mapUser(userMap.get(batch.createdBy), allowedUserFields)
+      };
+    });
+  }
+
+  async mapImportBatch(batch, user) {
+    const users = await this.userService.getUsersByIds([batch.createdBy]);
+    const allowedUserFields = privateData.getAllowedUserFields(user);
+
+    if (users.length !== 1) {
```

Comment:
Can this actually ever occur, since the users collection cannot have duplicate `_id`s?

Reply:
It's more about not having the user :)

Reply:
True, we always check in the mapper if the number matches, so it doesn't hurt to have it here.